### ISSUE TITLE
capnp fails to build (downloading gtest fails)

### ISSUE THREAD

**Issue by dfelicia** [Homebrew/homebrew-core#5435]
If Homebrew was updated on Aug 10-11th 2016 and `brew update` always says `Already up-to-date.` you need to run: `cd "$(brew --repo)" && git fetch && git reset --hard origin/master && brew update`.
# Please follow the general troubleshooting steps first:
- [X ] Ran `brew update` and retried your prior step?
- [ X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ X] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

_You can erase any parts of this template not applicable to your Issue._
### Bug reports:

capnp fails to build (downloading gtest fails)

https://gist.github.com/66fb76aa836e1b14e04d30b3d2b2c7ea

**Comment by zmwangx** [comment]
```diff
diff --git a/Formula/capnp.rb b/Formula/capnp.rb
index bfbd112..568691a 100644
--- a/Formula/capnp.rb
+++ b/Formula/capnp.rb
@@ -18,6 +18,15 @@ class Capnp < Formula
   def install
     ENV.cxx11
 
+    # Remove when the next release comes out
+    inreplace "src/CMakeLists.txt",
+      "http://googletest.googlecode.com/files/gtest-1.7.0.zip",
+      "https://github.com/google/googletest/archive/release-1.7.0.tar.gz"
+
+    inreplace "src/CMakeLists.txt",
+      "URL_MD5 2d6ec8ccdf5c46b05ba54a9fd1d130d7",
+      "URL_HASH SHA256=f73a6546fdf9fce9ff93a5015e0333a8af3062a152a9ad6bcb772c96687016cc"
+
     mkdir "build" do
       system "cmake", "..", *std_cmake_args
       system "make", "check"
```

Even then `make check` would fail:

```
2 - kj-heavy-tests-run (Failed)
```

**Comment by dunn** [comment]
We should add `gtest` as a resource, instead of downloading it during the installation.  If you'd like to try a PR for that, search through existing formula for `resource do`.

The test failure should be reported upstream, also.

**Comment by dunn** [comment]
@zmwangx `make check` passes for me locally, but I have to allow the test server to accept incoming connections.

**Comment by zmwangx** [comment]
@dunn I recall getting that failure even with system firewall completely off. But it works for you — good.

> The test failure should be reported upstream, also.

Upstream has moved on, gtest has been yanked out completely. We're dealing with a legacy release here. Probably worth asking them to cut a new release.

**Comment by paulproteus** [comment]
Hi nice homebrew people!

I'd love to see this get fixed. I'm a contributor to capnproto, and I wanted to just say that I'd love to see this build get fixed.

I agree that we've moved on from gtest, but we're a little while away from being able to do a new release. We're in the process of merging some fixes for Windows support. I know that this doesn't affect `brew` users directly, but having each release work on all our major platforms helps us limit how many users get confused. Therefore my hope is that I can convince y'all to fix 0.5.3 to work on homebrew while waiting for a new capnproto release.

Let me say here that I'm:

- +1 on the Oct 1 patch by @zmwangx , which allows capnproto to build even though Google broke our download URL.

- +1, for now, patching-out the test that requires allowing the capnproto server to listen locally. Since allowing that seems to cause problems for homebrew users, it's not worthwhile to require that test to pass IMHO.

- +1 on not bothering to add `gtest` as a resource, since in the next release it won't be needed anyway.

Let me know if that's useful input.

I also want to give my heartfelt thanks for all the work that everyone in the Homebrew community does to make software easy for people to use. Projects like Homebrew are fantastically important, and library authors like us sometimes forget how essential they are in allowing developers to benefit from the code we write.

So --  :bowing_man: thank you! If there's any other input I can give, please let me know.

**Comment by zmwangx** [comment]
@paulproteus There's already #5773, blocked due to the test failure. We can probably ship that if `make check` can be removed.
